find_email_addresses.py

```diff
@@ -0,0 +1,66 @@
+# Uses the following libraries:
+#   bs4 to find new links
+#   re to create regular expression for emails
+#   sys to get command line arguments
+#   time to wait until HTML Document is fully loaded
+#   deque to store which URLs to visit and traverse page breadth-first
+#   selenium to receive HTML response
+#   collections to store visited URLs and discovered emails
+import bs4, re, sys, time
+from selenium import webdriver 
+from collections import deque
+
+class EmailScraper():
+  EMAIL_REGEX = re.compile(r'[\w+\-.]+@[a-z\d\-.]+\.[a-z]+')
+  DELAY = 5
+
+  def __init__(self, base_url):
+    self.base_url = self.__format_url(base_url)
+    self.to_visit = deque([self.base_url])
+    self.visited = set()
+    self.unique_emails = set()
+    self.driver = webdriver.Firefox()
+
+  def run(self):
+    print("Found the following emails:")
+
+    while len(self.to_visit) > 0:
+      current_url = self.to_visit.popleft()
+
+      self.driver.get(current_url)
+      time.sleep(EmailScraper.DELAY) # Wait for page to load
+      soup = bs4.BeautifulSoup(self.driver.page_source)
+
+      self.__parse_emails(soup)
+      self.__get_child_urls(soup)
+
+    self.driver.close()
+
+  def __get_child_urls(self, soup):
```

Comment:
we tend to avoid double unders for private methods because it does a lot of other weird name mangling in debuggers and other stuff, so we just go with single underscore like `_get_child_urls`.  not a big deal, just commenting as if it were a real PR into our codebase.